### ISSUE TITLE
Bug: CLI fails to process glob consistently (TSConfig does not include this file...)

### ISSUE THREAD

**Issue by erikschul** [typescript-eslint/typescript-eslint#8269]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```TypeScript
This doesn't work, specifically for svelte files.

npx eslint --ext .astro,.svelte,.ts,.js src/
npx eslint "src/**/*"
npx eslint "src/**/*.{svelte,astro}"
```

which throws error
```
/workspace/src/components/svelte/Timer.svelte
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/src/components/svelte/Timer.svelte` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

This works
```
npx eslint src/components/svelte/*.svelte
npx eslint "src/**/*.svelte" "src/**/*.ts" "src/**/*.astro" "src/**/*.js"
```

### Actual Result

error

### Versions

Node version: 18.19.0
npm version: 10.2.3
Local ESLint version: 8.56.0
Global ESLint version: 8.56.0
Operating System: linux

**Comment by bradzacher** [comment]
Please use the "Complex Bug With a Reproduction" template and fill out with a reproduction so that we can investigate and reproduce your error to help you.
